scripts/check-code-formatting

```diff
@@ -2,7 +2,8 @@
 set -e
 
 # Ensure we are in root directory
-basedir="$(readlink -f `dirname $0`/..)"
+readlink_bin=$([[ "$OSTYPE" == "darwin"* ]] && printf greadlink || printf readlink)
```

Comment:
`greadlink` is not present on macOS by default. Instead, it is supplied by e.g. Homebrew. I think that, instead, we should check whether `greadlink` exists and if not, use `readlink`. However, this is outside of the scope for this PR.

Reply:
I don't think this change belongs in this PR at all. It has nothing to do with the feature/enhancement at hand.

Reply:
That's what I said..?

Reply:
I don't think you were being clear at all. What you said could easily be interpreted as "leave this for now and the greadlink/readlink stuff can be handled in another PR" (which is the way that I interpreted it).